### PR TITLE
add steps for azd automation

### PR DESCRIPTION
**Pull request by sonwan2020**
## Purpose
And fix doc order issue

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: infra/bicep/main.bicep [hunk 1]
```diff
@@ -29,36 +29,36 @@ param sqlAdmin string = 'sqladmin'
 param sqlAdminPassword string
 
 @description('Repo url of the configure server.')
-param configGitRepo string
+param configGitRepo string = 'https://github.com/Azure-Samples/java-microservices-aca-lab'
```
[the remark this answers is not in the document]
Comment (vermegi) [reply infra/bicep/main.bicep:32]: nvm, I see a configGitPath ... however, this is still not the config repo from the learner, config will not be correct.

File: docs/02_lab_migrate/0203.md [hunk 2]
```diff
@@ -70,7 +70,7 @@ Your MySQL database will also have a firewall enabled. This firewall will by def
    * It adds extra `eureka` config on lines 61 to 66.
    * It removes the `chaos-monkey` and `mysql` profiles.
 
-1. In the part you pasted, update the values of the target datasource endpoint on line 12, the corresponding admin user account on line 13, and its password on line 14 to match your configuration. Set these values by using the information in the Azure Database for MySQL Flexible Server connection string you recorded earlier in this task.
+1. In the part you pasted, update the values of the target datasource endpoint on line 6, the corresponding admin user account on line 7, and its password on line 8 to match your configuration. Set these values by using the information in the Azure Database for MySQL Flexible Server connection string you recorded earlier in this task.
```
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/02_lab_migrate/0203.md:73]: tested, and we are planning to replace the line number indicators to some comments string indicators

File: docs/06_lab_automation/0601.md [hunk 3]
```diff
@@ -1,63 +1,67 @@
...
-Fill the configuration file according to your environment.
+Prepare your local environment and plan your Azure resource, get ready for the one-click deployment.
+
+{: .note }
+> This automation guide is tested in linux environment only. Here we use WSL environment to run the automation tool.
 
 ## Step by step guidance
 
-1. If you want to use azd to deploy from a new environment, prepare your images in Azure Container Registry
+1. Install WSL environmet
```
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/06_lab_automation/0601.md:17]: fixed

File: docs/06_lab_automation/0601.md [hunk 4]
```diff
@@ -1,63 +1,67 @@
...
-    RESOURCE_GROUP=rg-$APPNAME
+1. Install tools
+
+   - Follow this guide to install azd tool to your WSL environment: [Install Azd](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/install-azd?tabs=winget-windows%2Cbrew-mac%2Cscript-linux&pivots=os-linux)
+   - Install az cli: [Install az](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=apt#option-1-install-with-one-command)
+   - Install docker: [Docker Desktop WSL 2](https://docs.docker.com/desktop/wsl/)
+
+1. Plan your Azure Container Registry
+
+   An Azure Container Registry is required to save the images for the solution, create new or use exist:
```
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/06_lab_automation/0601.md:29]: updated, thanks

File: docs/06_lab_automation/0601.md [hunk 5]
```diff
@@ -1,63 +1,67 @@
...
+   - Option 1: Create new
```
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/06_lab_automation/0601.md:30]: updated, thanks

File: docs/06_lab_automation/0601.md [hunk 6]
```diff
@@ -1,63 +1,67 @@
...
-    ACR_UMID=$(az identity show --resource-group $RESOURCE_GROUP --name $ACR_IDENTITY --query id --output tsv)
-    SP_ID=$(az identity show --resource-group $RESOURCE_GROUP --name $ACR_IDENTITY --query principalId --output tsv)
-    CLIENT_ID=$(az identity show --resource-group $RESOURCE_GROUP --name $ACR_IDENTITY --query clientId --output tsv)
+    - login to acr with command `az acr login -n $ACR_NAME`
+    - use a simple java image as the placeholder image:
 
-    # assign role
-    az role assignment create --assignee $SP_ID --scope $ACR_ID --role acrpull
+    ```bash
+    docker pull mcr.microsoft.com/azurespringapps/default-banner:distroless-2024022107-66ea1a62-87936983
```
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/06_lab_automation/0601.md:56]: This is the first version, in my plan, we use placeholder images to make sure 'resource provision' is fine
And the next we will 'azd deploy' the solution images to the provisioned services.

use image from mcr? There will be two kinds of [RegistryConfidential](https://learn.microsoft.com/en-us/azure/templates/microsoft.app/containerapps?pivots=deployment-language-bicep#registrycredentials)
- mcr: anonymous
- private cr: UMI
And we only use of the config only at any moment

Let's think about the improvements next

File: docs/06_lab_automation/0601.md [hunk 7]
```diff
@@ -1,63 +1,67 @@
...
+    docker tag mcr.microsoft.com/azurespringapps/default-banner:distroless-2024022107-66ea1a62-87936983 $ACR_NAME.azurecr.io/azurespringapps/default-banner:latest
+    docker push $ACR_NAME.azurecr.io/azurespringapps/default-banner:latest
```
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/06_lab_automation/0601.md:58]: good suggestion, I will try

File: docs/06_lab_automation/0603.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+---
+title: '3. Triage the issues'
+layout: default
+nav_order: 3
+parent: 'Lab 6: Deploy to Azure automatically'
+---
+
+# 3. Triage the issues
```
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/06_lab_automation/0603.md:8]: maybe I can add a link in doc 0602

File: docs/06_lab_automation/0603.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+1. Failed to provision java component 'configserver'
+
+    ```text
+    ERROR: error deploying infrastructure: deploying to subscription:
+
+    Deployment Error Details:
+    JavaComponentOperationError: Failed to provision java component 'configserver'. Error details: Failed to create config map external-auth-config-map for JavaComponent configserver in k8se-system namespace. There will be no re-tries..
+    ```
+
+    > This is a temporary error in service, just ignore and retry.
```
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/06_lab_automation/0603.md:26]: Yes, it is a bug, and the service team is working on it

File: docs/06_lab_automation/06_openlab_automation.md [hunk 10]
```diff
@@ -14,14 +14,14 @@ has_children: true
...
-In this lab, we import a new tool [azd](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/) to help deploy the solution to Azure with single command.
+In this lab, we introduce a new tool [azd](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/) to help deploy the solution to Azure with single command.
 
 ## Objectives
 
 After you complete this lab, you will be able to:
 
 - Get familiar with the azd tool.
-- Deploy the petclinic solution to new ACA environment.
+- Deploy the petclinic solution (with AI) to new ACA environment.
```
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/06_lab_automation/06_openlab_automation.md:24]: Yes, there are lots of options for different resources.

File: docs/02_lab_migrate/0203.md [hunk 11]
```diff
@@ -59,7 +59,7 @@ Your MySQL database will also have a firewall enabled. This firewall will by def
         --rule-name allAzureIPs \
         --name ${MYSQL_SERVER_NAME} \
         --resource-group ${RESOURCE_GROUP} \
-        --start-ip-address 0.0.0.0 --end-ip-address 0.0.0.0
+        --start-ip-address 0.0.0.0 --end-ip-address 255.255.255.255
```
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/02_lab_migrate/0203.md:62]: if both start & end address is 0.0.0.0, no items will be created for the firewall settings
Here for the public endpoint visit, we just allow all external IPs

File: docs/06_lab_automation/0601.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (sonwan2020) [reply docs/06_lab_automation/0601.md:58]: updated

File: infra/bicep/main.bicep (hunk 1 above)
[the remark this answers is not in the document]
Comment (sonwan2020) [reply infra/bicep/main.bicep:32]: This is for azd automation, we use the default settings to make the solution work.

**Review by haoozhang — APPROVED** [review]

**Pull request merged**